frontend/views/containers/proposals/Mincome.vue

```diff
@@ -50,33 +54,64 @@ export default {
         steps: [
           'GroupMincome'
         ]
-      },
-      fakeStore: {
-        groupCurrency: '$ USD'
       }
     }
   },
   validations: {
     form: {
-      incomeProvided: {
+      mincomeAmount: {
         required,
+        minValue: value => value > 0,
         decimals: decimals(2)
       }
     },
     // validation groups by route name for steps
     steps: {
       GroupMincome: [
-        'form.incomeProvided'
+        'form.mincomeAmount'
       ]
     }
   },
+  computed: {
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'groupShouldPropose',
+      'groupSettings'
+    ]),
+    inputSuffix () {
+      return `${currencies[this.groupSettings.incomeCurrency]} ${this.groupSettings.incomeCurrency}`
+    },
+    friendlyIncome () {
+      return `${currencies[this.groupSettings.incomeCurrency]}${this.groupSettings.mincomeAmount}`
+    }
+  },
   methods: {
-    submit (form) {
-      console.log(
-        'TODO: Logic to Propose Mincome.',
-        'mincome:', this.form.incomeProvided,
-        'reason:', form.reason
-      )
+    async submit (form) {
+      this.ephemeral.errorMsg = null
+
+      if (this.groupShouldPropose) {
+        return console.log(
+          'TODO: Logic to Propose Mincome.',
+          'mincome:', this.form.mincomeAmount,
+          'reason:', form.reason
+        )
+      }
+
+      try {
+        const updatedSettings = await sbp(
+          'gi.contracts/group/updateSettings/create',
+          // to avoid numbers with leading zeros (ex: 01)
+          { mincomeAmount: parseInt(this.form.mincomeAmount, 10) },
+          this.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', updatedSettings)
+        this.$refs.proposal.close()
+      } catch (error) {
+        console.log('ora', error)
```

Comment:
1. `'ora'` isn't a descriptive error message
2. errors must be logged with `console.error`

Reply:
improved